mediaflow_proxy/utils/http_utils.py

```
@@ -39,278 +30,149 @@ def __init__(self, status_code, message):
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error {e.response.status_code} while creating streaming response")
+            if e.response.status_code == 404:
+                logger.error(f"Segment Resource not found: {url}")
+                raise e
+            raise DownloadError(
+                e.response.status_code, f"HTTP error {e.response.status_code} while creating streaming response"
+            )
+        except httpx.RequestError as e:
             logger.error(f"Error creating streaming response: {e}")
             raise DownloadError(502, f"Error creating streaming response: {e}")
         except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)